src/lksearch/MASTSearch.py

```diff
@@ -1115,53 +1128,64 @@ def _download_one(
 
         # We don't want to query cloud_uri if we don't have to
         # First check to see if we're not downloading on a cloud platform
-        # If not - cloud_uris should have already been queried - in that case
+        # If so - cloud_uris should have already been queried - in that case
         # check to see if a cloud_uri exists, if so we just pass that
 
-        download = True
+        skip_download = np.zeros_like(self.table.obs_id).astype(dtype=bool)
```

Comment:
Such a nitpick here but zeros_like is designed to give you something with the same shape and type of the input so this might be more clear

```suggestion
        skip_download = np.zeros(len(self.table.obs_id), dtype=bool)
```